src/tools/build-manifest/src/main.rs

```diff
@@ -107,6 +107,7 @@ static MINGW: &'static [&'static str] = &[
 struct Manifest {
     manifest_version: String,
     date: String,
+    git_commit_hash: String,
```

Comment:
I think that this commit may be best inside each `Package`? I'd imagine that maybe eventually we'd have a use for the commit information other than just for `rustc` but also for `cargo`

Reply:
Everything under https://s3.amazonaws.com/rust-lang-ci/rustc-builds are in "subdirectories" named after the rust-lang/rust commit hash, even for other packages like Cargo. But if you prefer I could work on that change in case the Cargo commit hash is needed later for some reason.

Reply:
Figuring out the respective build systems in respective repositories to add text files in various tarballs to make this info available in the first place is non-trivial, though.

Reply:
I’ve made it optional (always missing at the moment for Cargo and RLS) and per-package.